File: Assignment/Services/DOT/DOTNode.cs
using System;

namespace Assignment.Services.DOT
{
    public struct DOTNode
    {
        public readonly string Lable;
        public readonly Guid Id;

        public DOTNode(string label)
        {
            Lable = label;
            Id = Guid.NewGuid();
        }

        public override string ToString()
        {
            return $"{Id}";
        }
    }
}

File: src/main/java/ce305/abstraction/dot/DotNode.java
package ce305.abstraction.dot;

import java.util.UUID;

public class DotNode {
    public final String lable;
    public final UUID id;

    public DotNode(String label) {
        this.lable = label;
        this.id = UUID.randomUUID();
    }

    @Override
    public String toString() {
        return this.id.toString();
    }

    @Override
    public boolean equals(Object obj) {
        return this.id.equals(((DotNode)obj).id);
    }
}